### PR TITLE
Add conan support

### PR DESCRIPTION
**Pull request by Lucas-Vertrees-Bose**

### CODE REVIEW

**Comment by Mark-Hatch-Bose** [comment]
Still requires `ConanBuild` github status. Needs webhook for Jenkins build.
Otherwise the changes look good.

Also for all these repos, we should confirm that EF output is not changed, and conan build matches EF output. Thanks!

**Comment by Mark-Hatch-Bose** [comment]
Still not seeing a webhook here for conan build, do not merge without a passing build.

**Comment by Derek-Richardson-Bose** [comment]
@Mark-Hatch-Bose pushed white space change and it seems to go now

**Comment by Mark-Hatch-Bose** [comment]
Packaging looks good in Jenkins
```
opensource-curl/7.54.1-5+gcb90e6d@BoseCorp/PR-3: Calling package()
opensource-curl/7.54.1-5+gcb90e6d@BoseCorp/PR-3 package(): Packaged 1 '.gz' file: component-info.gz
opensource-curl/7.54.1-5+gcb90e6d@BoseCorp/PR-3 package(): Packaged 2 files: curl, curl-config
opensource-curl/7.54.1-5+gcb90e6d@BoseCorp/PR-3 package(): Packaged 10 '.h' files
opensource-curl/7.54.1-5+gcb90e6d@BoseCorp/PR-3 package(): Packaged 1 '.m4' file: libcurl.m4
opensource-curl/7.54.1-5+gcb90e6d@BoseCorp/PR-3 package(): Packaged 1 '.la' file: libcurl.la
opensource-curl/7.54.1-5+gcb90e6d@BoseCorp/PR-3 package(): Packaged 1 '.pc' file: libcurl.pc
opensource-curl/7.54.1-5+gcb90e6d@BoseCorp/PR-3 package(): Packaged 1 '.a' file: libcurl.a
```
Should/have we compare the output packages from EF and Conan to ensure they are the same?

**Comment by Derek-Richardson-Bose** [comment]
We compared them with by building using script and with conan and libs both had same MD5 sums @Mark-Hatch-Bose

File: Bose/build-qc8017_32 [hunk 1]
```diff
@@ -9,14 +9,18 @@ if (( $# != 0 )); then
     exit 1
 fi
 
-cd .. || exit
+if [ ! -f "configure.ac" ]; then
+  cd ..
```
Comment (Michael-Cook-Bose) [inline Bose/build-qc8017_32]: Please fix the indentation. Here and througout. Each indentation level should be 4 spaces.

File: Bose/build-qc8017_32 [hunk 2]
```diff
@@ -9,14 +9,18 @@ if (( $# != 0 )); then
     exit 1
 fi
 
-cd .. || exit
+if [ ! -f "configure.ac" ]; then
```
Comment (Michael-Cook-Bose) [inline Bose/build-qc8017_32]: The double quotes here have no effect and would tend to make the reader wonder if the author knew what he was doing?

File: Bose/build-qc8017_32 [hunk 3]
```diff
@@ -77,9 +80,14 @@ rm -rf "$build_dir/share/man"
 
 components record "$build_dir" || exit
 
-cd "$build_dir/.." || exit
-base=$(basename "$build_dir")
-tar -cjf "$base.tar.bz2" "$base" || exit
-echo "Wrote $PWD/$base.tar.bz2"
+if [ -z "$DONT_TAR" ]
+then
```
Comment (Michael-Cook-Bose) [inline Bose/build-qc8017_32]: Move to previous line per the usual style.
`if [ -z "$DONT_TAR" ]; then`

File: Bose/build-qc8017_64 [hunk 4]
```diff
@@ -9,14 +9,18 @@ if (( $# != 0 )); then
...
+  cd ..
+fi
 if ! [ -e configure ]; then
     ./buildconf || exit
 fi
 if [ -e Makefile ]; then
     make clean || exit
 fi
-cd Bose/ || exit
+if [ -d "./Bose" ]; then
```
Comment (Michael-Cook-Bose) [inline Bose/build-qc8017_64]: What is this about? Why wouldn't the Bose directory exist?

**Comment by Lucas-Vertrees-Bose** [comment]
@Michael-Cook-Bose My original thinking was to run the scripts inside the base folder, which is why I checked for `configure.ac` and the `Bose` folder to `cd` to the right directory. Now I just cd into the `Bose` folder from the get go before running the scripts.

File: Bose/build-qc8017_32 [hunk 5]
```diff
@@ -16,7 +16,7 @@ fi
 if [ -e Makefile ]; then
     make clean || exit
 fi
-cd Bose/ || exit
+cd ./Bose
```
Comment (Michael-Cook-Bose) [inline Bose/build-qc8017_32]: Revert please.

File: Bose/build-qc8017_32 [hunk 6]
```diff
@@ -37,7 +37,6 @@ CXX="$bin/$arch-g++ -fPIC -march=armv7-a -mfloat-abi=softfp -mfpu=neon -mtune=co
 
 LDFLAGS="-Wl,-R,'\$\$ORIGIN/../lib'"
 
-cd "$(dirname "$0")" || exit
```
Comment (Michael-Cook-Bose) [inline Bose/build-qc8017_32]: Why this change?

File: Bose/build-qc8017_32 [hunk 7]
```diff
@@ -16,7 +16,7 @@ fi
 if [ -e Makefile ]; then
     make clean || exit
 fi
-cd Bose/ || exit
+cd Bose/ || exit
```
Comment (Michael-Cook-Bose) [inline Bose/build-qc8017_32]: Please revert.

File: Bose/build-qc8017_64 [hunk 8]
```diff
@@ -16,7 +16,7 @@ fi
 if [ -e Makefile ]; then
     make clean || exit
 fi
-cd Bose/ || exit
+cd Bose/ || exit
```
Comment (Michael-Cook-Bose) [inline Bose/build-qc8017_64]: Please revert.

File: Bose/build-x86 [hunk 9]
```diff
@@ -16,7 +16,7 @@ fi
 if [ -e Makefile ]; then
     make clean || exit
 fi
-cd Bose/ || exit
+cd Bose/ || exit
```
Comment (Michael-Cook-Bose) [inline Bose/build-x86]: Please revert.

File: Bose/build-x86_64 [hunk 10]
```diff
@@ -9,14 +9,14 @@ if (( $# != 0 )); then
     exit 1
 fi
 
-cd .. || exit
+cd ..
```
Comment (Michael-Cook-Bose) [inline Bose/build-x86_64]: Please revert.

File: Bose/build-x86_64 [hunk 11]
```diff
@@ -9,14 +9,14 @@ if (( $# != 0 )); then
...
 if ! [ -e configure ]; then
     ./buildconf || exit
 fi
 if [ -e Makefile ]; then
     make clean || exit
 fi
-cd Bose/ || exit
+cd Bose/ || exit
```
Comment (Michael-Cook-Bose) [inline Bose/build-x86_64]: Please revert.
